api/helpers.go

```diff
@@ -0,0 +1,16 @@
+package api
+
+import (
+	"encoding/json"
+	"io"
+)
+
+func decodeJSON(obj interface{}, body io.ReadCloser) error {
```

Comment:
nit: Probably we can just take `io.Reader` here as we do not use `Close()`.
```suggestion
func decodeJSON(obj interface{}, body io.Reader) error {
```